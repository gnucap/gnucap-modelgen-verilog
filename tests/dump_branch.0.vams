
`include "sub/disciplines.few"

module branchtest(a, b);
	electrical a, b, c;
	branch (a, b) mybr1, mybr1_alias;
	branch (c, b) mybr2;
	branch (b, c) mybr3;

	analog begin
		I(a,b) <+ 1;
		I(b,a) <+ 2;
		I(mybr2) <+ 3;
		I(mybr3) <+ 4;
	end
endmodule
