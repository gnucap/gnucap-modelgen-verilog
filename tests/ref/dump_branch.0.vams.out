nature Current
  units = "A";
  access = "I";
  idt_nature = "Charge";
  abstol = "1e-12";
endnature

nature Charge
  units = "coul";
  access = "Q";
  ddt_nature = "Current";
  abstol = "1e-14";
endnature

nature Voltage
  units = "V";
  access = "V";
  idt_nature = "Flux";
  abstol = "1e-6";
endnature


discipline electrical
  potential "Voltage";
  flow "Current";
enddiscipline


module branchtest(a, b);
  electrical a, b, c;
  branch (a, b) mybr1, mybr1_alias;
  branch (c, b) mybr2;
  branch (b, c) mybr3;
  analog begin
    I(a, b) <+ 1;
    I(b, a) <+ 2;
    I(mybr2) <+ 3;
    I(mybr3) <+ 4;
  end

endmodule



